agent/dnsx_agent.py

```diff
@@ -42,40 +40,71 @@ def process(self, message: m.Message) -> None:
             message:
         """
         domain = message.data['name']
+        wordlist = self.args.get('wordlist')
         logger.info('scanning domain %s', domain)
-        if not self.set_add('agent_dnsx_asset', domain):
+        if not self.set_add(b'agent_dnsx_asset', domain):
             logger.info('target %s/ was processed before, exiting', domain)
             return
 
-        results = self._run_dnsx(domain)
+        results = self._run_dnsx_resolve(domain)
         if results is not None:
             self._emit_results(domain, results)
 
-    def _emit_results(self, domain: str, results: Dict) -> None:
+        if wordlist is not None:
+            results = self._run_dnsx(domain, wordlist)
+            if results is not None:
+                self._emit_results(domain, results)
+
+    def _emit_results(self, domain: str, results: List) -> None:
         """Parses results and emits records."""
         for record in result_parser.parse_results(results):
             logger.info('emitting result for %s', record)
             self.emit(selector='v3.asset.domain_name.dns_record',
                       data={'name': domain, 'record': record.record, 'values': record.value})
 
-    def _run_dnsx(self, domain: str):
+    def _run_dnsx(self, domain: str, wordlist: Optional[str] = None):
         """Run dnsx and returns the results."""
-        with tempfile.NamedTemporaryFile() as input_domain,\
-                tempfile.NamedTemporaryFile() as output_domain:
+        command = self._prepare_command(domain, wordlist)
+        logger.info('running command %s', command)
+        result = subprocess.run(command, capture_output=True, check=False)
+        if result.returncode == 0 and result.stdout != b'':
+            return [json.loads(l) for l in result.stdout.decode().split('\n') if l != '']
+        else:
+            logger.warning('Empty result file for domain %s', domain)
+
+    def _prepare_command(self, domain, wordlist: Optional[str]) -> List[str]:
+        """Prepare dnsx command."""
+        command = ['dnsx',
+                   '-silent',
+                   '-a', '-aaaa', '-cname', '-ns', '-txt', '-ptr', '-mx', '-soa', '-resp',
+                   '-json',
+                   '-d', domain]
+
+        if wordlist is not None:
+            command.extend(['-w', wordlist])
+        return command
+
+    def _run_dnsx_resolve(self, domain: str):
+        """Run dnsx and returns the results."""
+        with tempfile.NamedTemporaryFile() as input_domain:
             input_domain.write(domain.encode())
             input_domain.flush()
-            command = self._prepare_command(str(pathlib.Path(input_domain.name)), str(pathlib.Path(output_domain.name)))
+            command = self._prepare_command_resolve(input_domain.name)
```

Comment:
you can set this code [line 92 , line 98] in a separate function and call it from both _run_dnsx_resolve and  _run_dnsx